Objects/frameobject.c

```diff
@@ -1197,102 +1805,52 @@ frame_get_var(_PyInterpreterFrame *frame, PyCodeObject *co, int i,
 
 
 PyObject *
-_PyFrame_GetLocals(_PyInterpreterFrame *frame, int include_hidden)
+_PyFrame_GetLocals(_PyInterpreterFrame *frame)
 {
-    /* Merge fast locals into f->f_locals */
-    PyObject *locals = frame->f_locals;
-    if (locals == NULL) {
-        locals = frame->f_locals = PyDict_New();
-        if (locals == NULL) {
-            return NULL;
-        }
+    PyCodeObject* co = _PyFrame_GetCode(frame);
+    PyFrameObject* f = _PyFrame_GetFrameObject(frame);
+
+    if (!(co->co_flags & CO_OPTIMIZED)) {
+        return Py_NewRef(frame->f_locals);
     }
-    PyObject *hidden = NULL;
 
-    /* If include_hidden, "hidden" fast locals (from inlined comprehensions in
-       module/class scopes) will be included in the returned dict, but not in
-       frame->f_locals; the returned dict will be a modified copy. Non-hidden
-       locals will still be updated in frame->f_locals. */
-    if (include_hidden) {
-        hidden = PyDict_New();
-        if (hidden == NULL) {
-            return NULL;
-        }
+    return _PyFrameLocalsProxy_New(f);
+}
+
+
+PyObject*
+_PyFrame_GetHiddenLocals(_PyInterpreterFrame *frame)
```

Comment:
Could you add a comment explaining what "hidden" locals are, for those unfamiliar with PEP 709 (if that is the reason for them).